### PR TITLE
Sync with Upstream on Jan. 2, 2017

### PR DESCRIPTION
**Pull request by pjdufour**
Variety of changes to catch up with GeoNode upstream master.

### CODE REVIEW

File: project_name/settings.py [hunk 1]
```diff
@@ -22,8 +22,8 @@
 import os
 from kombu import Queue
 from geonode import __file__ as geonode_path
-from geonode import get_version
-from geonode.celery_app import app  # flake8: noqa
+from {{ project_name }} import get_version
```
Comment (jj0hns0n) [inline project_name/settings.py]: This wont work, it needs to be from geonode

**Review by simod — COMMENTED** [review]
Why we need all these files? pavement, .gitignore, yaml files etc?

**Review by simod — COMMENTED** [review]
I'm also -1 on having all these settings, we should just inherit from geonode as we were doing.

**Comment by jj0hns0n** [comment]
I also think it makes more sense to have as _minimal_ a set of settings in this project as necessary and inherit from GeoNode as much as possible. @pjdufour can we pare this back to whatever is minimally required?

**Comment by pjdufour** [comment]
**New Files**
The `pavement.py`, `dev_config.yml`, etc. is to enable the same dev workflow on downstream projects that we have on upstream (`paver setup`, `paver stop`, etc.).  Pavement.py uses `{{ project_name }}` to be properly configured.  For a `dev_config.yml` in this repo, your artifacts might be at a custom location.  New devs shouldn't need to edit `pavement.py` code to get up and running, but just edit the yaml file.  I don't personally use docker, so not sure if those are necessary.

**Settings**
On some downstream projects I previously worked on (ROGUE, etc.), we did have a `from geonode.settings import *`; however, those were very heavily customized instances.  The `settings.py` file in this `geonode-project` repo itself was previously "flat" and did not import from `geonode.settings`.

With the move to use environmental variables and the upstream `geonode` package installed in site-packages, rather than as editable, we should be cautious with adding layers of imports into settings.

**Pull request merged**